<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Mitogenome Annotation Wrapper — annotate • MitoPilot</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Mitogenome Annotation Wrapper — annotate"><meta name="description" content="Uses Mitos2 and tRNAscan-SE to annotate a mitogenome assembly."><meta property="og:description" content="Uses Mitos2 and tRNAscan-SE to annotate a mitogenome assembly."></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">MitoPilot</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9000</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item"><a class="nav-link" href="../articles/index.html">Articles</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/Smithsonian/MitoPilot/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Mitogenome Annotation Wrapper</h1>
      <small class="dont-index">Source: <a href="https://github.com/Smithsonian/MitoPilot/blob/main/R/annotate.R" class="external-link"><code>R/annotate.R</code></a></small>
      <div class="d-none name"><code>annotate.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Uses Mitos2 and tRNAscan-SE to annotate a mitogenome assembly.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">annotate</span><span class="op">(</span></span>
<span>  assembly_fn <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  coverage_fn <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  cpus <span class="op">=</span> <span class="fl">4</span>,</span>
<span>  genetic_code <span class="op">=</span> <span class="st">"2"</span>,</span>
<span>  ref_db <span class="op">=</span> <span class="st">"Chordata"</span>,</span>
<span>  ref_dir <span class="op">=</span> <span class="st">"/home/harpua/Jzonah/MitoPilot/ref_dbs/Mitos2"</span>,</span>
<span>  mitos_opts <span class="op">=</span> <span class="st">"--intron 0 --oril 0 --trna 0"</span>,</span>
<span>  mitos_condaenv <span class="op">=</span> <span class="st">"mitos"</span>,</span>
<span>  trnaScan_opts <span class="op">=</span> <span class="st">"-M vert"</span>,</span>
<span>  trnaScan_condaenv <span class="op">=</span> <span class="st">"base"</span>,</span>
<span>  start_gene <span class="op">=</span> <span class="st">"trnF"</span>,</span>
<span>  out_dir <span class="op">=</span> <span class="cn">NULL</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-assembly-fn">assembly_fn<a class="anchor" aria-label="anchor" href="#arg-assembly-fn"></a></dt>
<dd><p>Path to the mitogenome assembly FASTA file.</p></dd>


<dt id="arg-coverage-fn">coverage_fn<a class="anchor" aria-label="anchor" href="#arg-coverage-fn"></a></dt>
<dd><p>Path to the mitogenome assembly coverage stats CSV file.</p></dd>


<dt id="arg-cpus">cpus<a class="anchor" aria-label="anchor" href="#arg-cpus"></a></dt>
<dd><p>Number of CPUs to use.</p></dd>


<dt id="arg-genetic-code">genetic_code<a class="anchor" aria-label="anchor" href="#arg-genetic-code"></a></dt>
<dd><p>Genetic code to use for annotation (default: 2).</p></dd>


<dt id="arg-ref-db">ref_db<a class="anchor" aria-label="anchor" href="#arg-ref-db"></a></dt>
<dd><p>Reference Mitos2 database to use for annotation (default:
"Chordata").</p></dd>


<dt id="arg-ref-dir">ref_dir<a class="anchor" aria-label="anchor" href="#arg-ref-dir"></a></dt>
<dd><p>Path to the Mitos2 reference database.</p></dd>


<dt id="arg-mitos-opts">mitos_opts<a class="anchor" aria-label="anchor" href="#arg-mitos-opts"></a></dt>
<dd><p>Additional command line options for MITOS2.</p></dd>


<dt id="arg-mitos-condaenv">mitos_condaenv<a class="anchor" aria-label="anchor" href="#arg-mitos-condaenv"></a></dt>
<dd><p>Conda environment to run MITOS2 (default: "mitos").</p></dd>


<dt id="arg-trnascan-opts">trnaScan_opts<a class="anchor" aria-label="anchor" href="#arg-trnascan-opts"></a></dt>
<dd><p>Additional command line options for tRNAscan-SE.</p></dd>


<dt id="arg-trnascan-condaenv">trnaScan_condaenv<a class="anchor" aria-label="anchor" href="#arg-trnascan-condaenv"></a></dt>
<dd><p>Conda environment to run tRNAscan-SE (default:
"base").</p></dd>


<dt id="arg-start-gene">start_gene<a class="anchor" aria-label="anchor" href="#arg-start-gene"></a></dt>
<dd><p>name of gene (PCG, rRNA, or tRNA) to start circular assembly (default = "trnF")</p></dd>


<dt id="arg-out-dir">out_dir<a class="anchor" aria-label="anchor" href="#arg-out-dir"></a></dt>
<dd><p>Output directory.</p></dd>

</dl></div>

  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Devin Leopold.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer></div>





  </body></html>

